### ISSUE TITLE
read weights from saved model

### ISSUE THREAD

**Issue by sun-peach** [keras-team/keras#12600]
Hi, I tried to use load_model() to load a model which was saved by the function model.save(). However, it reports "None gradient" error, since I have complicated network structure I defined. I tried model_from_yaml with load_weights. It worked very well.

Now, I get a model (say "model.h5") saved by model.save(). I have learnt that load_model() will cause problem, while load weights will be fine. Can I load its weights only?

**Comment by JamesTrick** [comment]
Hi @sun-peach - do you have any custom layers in your model? [If so, this may help](https://keras.io/getting-started/faq/#handling-custom-layers-or-other-custom-objects-in-saved-models).

**Comment by sun-peach** [comment]
@JamesTrick Thanks. Yes, I do use custom layers and I do use this approach to handle the custom layers. But it still shows None gradient error - That's why I turned to model_from_yaml approach (which works fine for me).

The problem is not reading the model in now - I have update all my current saving and loading approach with the yaml and save_weights functions. The problem is I have a model trained before (and saved with "model.save()" ). I want to read the weights from it (not load the entire network, just get the weights). How can I get the weights of the model which is saved with "model.save()"?

Thank you once again.
